### DISCUSSION TITLE
Importing MD files

### DISCUSSION THREAD

**Discussion by smccullan** [outline/outline#1605] [General]
Hi there

We wanted to see if the import export functionality was working. 

We exported the welcome pages. It gives us a zip file.

However when importing, the current setup does not permit importing of zip's and then building out the subfolder/page structure. Even more concerning is that the system cannot import MD files.

This makes the software very silo'd and makes us less keen on rolling out our data on the platform. 

Please update to allow md imports, and even better would be portability through the zip files building out the structure of the document.

**Comment by tommoor** [comment]
> Please update to allow md imports

This functionality already exists, I'm not sure what's making you think it doesn't. If you're seeing errors I'd recommend filing a bug with as much detail as possible – thanks!

**Comment by mrjazz** [comment]
For me it doesn't work. I do create Export for Welcome collection and trying to import it back in another collection. 
Server response with:
```
{"ok":false,"error":"invalid_request","status":400,"message":"File type application/zip not supported"}
```

**Reply by mrjazz** [reply]
Yes, as far as I see zip files aren't supported:

```javascript
// from documentImporter.js
const importMapping: ImportableFile[] = [
  {
    type: "application/msword",
    getMarkdown: confluenceToMarkdown,
  },
  {
    type:
      "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
    getMarkdown: docxToMarkdown,
  },
  {
    type: "text/html",
    getMarkdown: htmlToMarkdown,
  },
  {
    type: "text/plain",
    getMarkdown: fileToMarkdown,
  },
  {
    type: "text/markdown",
    getMarkdown: fileToMarkdown,
  },
];
```

**Reply by tommoor** [reply]
That's right, I would accept a PR that adds support for parsing an uploaded zip file. There are libraries already included for zip manipulation for the export so it probably wouldn't be a huge lift.

**Reply by mrjazz** [reply]
Great! Thanks. Looking forward to it.
